src/main/java/ServiceLayer/UserService.java

```diff
@@ -61,4 +64,22 @@ public Response register(String userName, String password, String email) {
         }
         return response;
     }
+
+    public Response purchaseCart(String token, List<Integer> busketsToBuy, PaymentMethod paymentMethod, SupplyMethod shippingMethod) {
+        Response response = new Response();
+        try {
+            String username = tokenService.getUsernameFromToken(token);
```

Comment:
notice the conversation in the group whatsup - you can also buy if you are a guest in the system. I think its important of the saving the orders for the history if he loggs in. also you did not mention the saving in history purchase for the user and in the store- you can write //TODO just so we dont forget its suppose to happened also

Reply:
Metar got the history task. assigned to him